src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509AuthorityKeyIdentifierExtension.cs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography.X509Certificates
+{
+    internal sealed class X509AuthorityKeyIdentifierExtension : X509Extension
+    {
+        private string _keyIdentifier;
+        private bool _hasComplexValue;
+        private bool _decoded;
+
+        internal X509AuthorityKeyIdentifierExtension()
+            : base(Oids.AuthorityKeyIdentifier)
+        {
+            _decoded = true;
+        }
+
+        internal string KeyIdentifier
+        {
+            get
+            {
+                if (!_decoded)
+                    Decode();
+
+                return _keyIdentifier;
+            }
+        }
+
+        internal bool HasComplexValue
```

Comment:
This could use a comment since complex currently means @bartonjs thinks it's complicated